vim-engine/src/main/kotlin/com/maddyhome/idea/vim/put/VimPutBase.kt

```diff
@@ -433,6 +434,11 @@ abstract class VimPutBase : VimPut {
         SelectionType.LINE_WISE -> {
           startOffset =
             min(vimEditor.text().length, injector.motion.moveCaretToLineEnd(vimEditor, line, true) + 1)
+          // At the end of a notebook cell the next symbol is a guard,
+          // so we add a newline to be able to paste. Fixes VIM-2577
+          if (vimEditor.document.getOffsetGuard(Offset(startOffset))!= null) {
+            application.runWriteAction { (vimEditor as MutableVimEditor).insertText((startOffset-1).offset, "\n") }
```

Comment:
Do we need a `startOffset > 0` check here?

Reply:
startOffset is only 0 if vimEditor.text().length is 0.
We only get in trouble if there is also a guard at that 0 position.
I think this case should not exist? But let's be safe instead of sorry.